### PR TITLE
Install pdb files

### PR DESCRIPTION
**Pull request by dschmidt**

### CODE REVIEW

File: bin/BuildSystem/BuildSystemBase.py [hunk 1]
```diff
@@ -240,18 +240,17 @@ def internalPostInstall(self):
 
             regexp = re.compile('{.*} (.*)')
             for f in files:
-                expectedPdb = os.path.splitext(f)[0] + '.pdb'
-                if not os.path.exists(expectedPdb):
+                if not os.path.exists(os.path.splitext(f)[0] + '.pdb'):
```
Comment (TheOneRing) [inline bin/BuildSystem/BuildSystemBase.py]: pleas use " quotes and format strings, either `f"{os.path.splitext(f)[0]}.pdb"` or `"{0}.pdb".format(os.path.splitext(f)[0])`

File: bin/BuildSystem/BuildSystemBase.py [hunk 2]
```diff
@@ -240,18 +240,17 @@ def internalPostInstall(self):
 
             regexp = re.compile('{.*} (.*)')
             for f in files:
```
Comment (TheOneRing) [inline bin/BuildSystem/BuildSystemBase.py:243]: `if CraftCore.cache.findApplication("peparser"):`

File: bin/BuildSystem/BuildSystemBase.py [hunk 3]
```diff
@@ -240,18 +240,17 @@ def internalPostInstall(self):
...
-                expectedPdb = os.path.splitext(f)[0] + '.pdb'
-                if not os.path.exists(expectedPdb):
+                if not os.path.exists(os.path.splitext(f)[0] + '.pdb'):
                     peparserOutput = CraftCore.cache.getCommandOutput(f"peparser", f"--pdb {f}")[1].strip()
                     pdbs = regexp.findall(peparserOutput)
+
+                    if not os.path.basename(f) == "icudt58.dll":
+                        assert len(pdbs) > 0, f'No pdb file available: {f}'
```
Comment (TheOneRing) [inline bin/BuildSystem/BuildSystemBase.py]: I'd prefer a warning and no special treatment for icu

**Review by TheOneRing — CHANGES_REQUESTED** [review]

File: bin/BuildSystem/BuildSystemBase.py (hunk 2 above)
Comment (dschmidt) [reply bin/BuildSystem/BuildSystemBase.py:243]: I have that check a few lines above?
-> https://github.com/KDE/craft/blob/b5d8444910e59e4ba0987353b581bbcdfa95a153/bin/BuildSystem/BuildSystemBase.py#L237

File: bin/BuildSystem/BuildSystemBase.py (hunk 1 above)
Comment (dschmidt) [reply bin/BuildSystem/BuildSystemBase.py]: Okay.

File: bin/BuildSystem/BuildSystemBase.py (hunk 3 above)
Comment (dschmidt) [reply bin/BuildSystem/BuildSystemBase.py]: I hate the special treatment for icu as well. I'd argue that if the pdb is missing it's an issue in the build system that should be fixed. And if there's only a warning we most certainly will miss it. If we find out that there are a lot of these issues and it's unreasonable to fix them all at this point, we could switch to a warning?

File: bin/BuildSystem/BuildSystemBase.py (hunk 2 above)
Comment (TheOneRing) [reply bin/BuildSystem/BuildSystemBase.py:243]: I see

File: bin/BuildSystem/BuildSystemBase.py (hunk 3 above)
Comment (TheOneRing) [reply bin/BuildSystem/BuildSystemBase.py]: then at least use a regexp to match :P 

I also expect the next full rebuild to be painful :D

Comment (dschmidt) [reply bin/BuildSystem/BuildSystemBase.py]: Hehehehe... likely :-\ 
But we can make it a list of exceptions then and take care of them one by one... 
Or we make it a warning and grep for it in the logs.. are we going to have a full log of the full rebuild?

Comment (dschmidt) [reply bin/BuildSystem/BuildSystemBase.py]: Made it a regexp

Comment (TheOneRing) [reply bin/BuildSystem/BuildSystemBase.py]: We keep the full logs of the last n builds.

Comment (dschmidt) [reply bin/BuildSystem/BuildSystemBase.py]: Okay made it a warning. 
Now I'm curious... :-D

File: bin/BuildSystem/BuildSystemBase.py [hunk 4]
```diff
@@ -232,4 +232,26 @@ def internalPostInstall(self):
...
+                                                 lambda x, root: True)
+
+            regexp = re.compile('{.*} (.*)')
+            exclude = re.compile(r'icudt[0-9]*.dll')
+            for f in files:
+                if not os.path.exists(f"{os.path.splitext(f)[0]}.pdb"):
+                    peparserOutput = CraftCore.cache.getCommandOutput("peparser", f"--pdb {f}")[1].strip()
+                    pdbs = regexp.findall(peparserOutput)
+
+                    if not len(pdbs) > 0 and not exclude.match(os.path.basename(f)):
```
Comment (TheOneRing) [inline bin/BuildSystem/BuildSystemBase.py]: just `if not pdbs and...`

**Review by TheOneRing — APPROVED** [review]

**Comment by TheOneRing** [comment]
merged

**Pull request closed without merging**